### ISSUE TITLE
Dynamic group filtering on tenant groups for content type devices not working correctly

### ISSUE THREAD

**Issue by Kircheneer** [nautobot/nautobot#2887]
### Environment
* Nautobot version / Python Version: 1.4.4/3.9, 1.4.10/3.9, 1.5.0/3.7 (demo env)
* Python version:
* Database platform, version: Postgres 14
* Middleware(s): None

### Steps to Reproduce
1. Create a tenant group "Test"
2. Create a tenant "Test"
3. Create a device "Test" and assign to it the tenant "Test"
4. Create a dynamic group "Test" with content type device
5. Edit the dynamic group and add a filter for tenant group equals "Test"

### Expected Behavior
The dynamic group called "Test" has 1 member device, namely the device called "Test" 

### Observed Behavior
1.4.4/1.4.10: The dynamic group has no members
1.5.0 (Nov 21 available on demo environment: https://demo.nautobot.com/extras/dynamic-groups/test/): 
```
<class 'django.core.exceptions.FieldDoesNotExist'>

Device has no field named 'tenant__group'

Python version: 3.7.15
Nautobot version: 1.5.0
```

**Comment by bryanculver** [comment]
@Kircheneer Can you confirm in 1.5.1? It seems similar to this: https://github.com/nautobot/nautobot/issues/2824

**Comment by Kircheneer** [comment]
Yes it has indeed been resolved with 1.5.1, sorry I should have checked with the latest version. Thanks!
